nipy/labs/viz_tools/maps_3d.py

```diff
@@ -117,14 +117,15 @@ def m2screenshot(mayavi_fig=None, mpl_axes=None, autocrop=True):
     if mpl_axes is not None:
         plt.axes(mpl_axes)
 
-    filename = tempfile.mktemp('.png')
+    fd, filename = tempfile.mkstemp('.png')
     mlab.savefig(filename, figure=mayavi_fig)
     image3d = plt.imread(filename)
     if autocrop:
         bg_color = mayavi_fig.scene.background
         image3d = autocrop_img(image3d, bg_color)
     plt.imshow(image3d)
     plt.axis('off')
+    os.close(fd)
     os.unlink(filename)
```

Comment:
Sorry, this is what I was proposing:

```suggestion
    image3d = mlab.screenshot()
    if autocrop:
        bg_color = mayavi_fig.scene.background
        image3d = autocrop_img(image3d, bg_color)
    plt.imshow(image3d)
    plt.axis('off')
```

Reply:
Alright, let me add these changes. It looks much cleaner.